platform/pulpcore/app/models/storage.py

```diff
@@ -225,176 +136,35 @@ def _save(path, content):
 
         return path
 
-    @staticmethod
-    def get_artifact_path(sha256digest):
-        """
-        Determine the absolute path where a file backing the Artifact should be stored.
-
-        Args:
-            sha256digest (str): sha256 digest of the file for the Artifact
-
-        Returns:
-            A string representing the absolute path where a file backing the Artifact should be
-            stored
-        """
-        return os.path.join(settings.MEDIA_ROOT, 'artifact', sha256digest[0:2], sha256digest[2:])
-
-    @staticmethod
-    def published_metadata_path(model, name):
-        """
-        Get the storage path for published metadata.
-
-        Args:
-            model (pulpcore.app.models.PublishedMetadata): A model instance.
-            name (str): The file name.
-
-        Returns:
-            str: The absolute storage path.
-        """
-        return os.path.join(
-            settings.MEDIA_ROOT,
-            'published',
-            'metadata',
-            str(model.pk),
-            name)
-
-    def get_available_name(self, name, max_length=None):
```

Comment:
This method is used to derive a new unique file-name for the path in cases where the file already exists.  We never want this to happen for artifacts.  To be safe, Let's override and raise an exception.